internal/bminventory/inventory.go

```diff
@@ -1733,7 +1736,47 @@ func (b *bareMetalInventory) updateHostsData(ctx context.Context, params install
 			return common.NewApiError(http.StatusConflict, err)
 		}
 	}
+	return nil
+}
+
+func (b *bareMetalInventory) updateHostsDiskSelection(ctx context.Context, params installer.UpdateClusterParams, db *gorm.DB, log logrus.FieldLogger) error {
+	for i := range params.ClusterUpdateParams.DisksSelectedConfig {
+		var host models.Host
+		err := db.First(&host, "id = ? and cluster_id = ?",
+			params.ClusterUpdateParams.DisksSelectedConfig[i].ID, params.ClusterID).Error
+		if err != nil {
+			log.WithError(err).Errorf("failed to find host <%s> in cluster <%s>",
+				params.ClusterUpdateParams.DisksSelectedConfig[i].ID, params.ClusterID)
+			return common.NewApiError(http.StatusNotFound, err)
+		}
+
+		var installationDiskPath = ""
+		for diskIdx := range params.ClusterUpdateParams.DisksSelectedConfig[i].DisksConfig {
```

Comment:
```suggestion
		for diskIdx, diskConfig := range params.ClusterUpdateParams.DisksSelectedConfig[i].DisksConfig {
```
then you will not need `diskConfig := params.ClusterUpdateParams.DisksSelectedConfig[i].DisksConfig[diskIdx]`